src/api/app/services/github_status_reporter.rb

```diff
@@ -0,0 +1,52 @@
+class GithubStatusReporter < SCMStatusReporter
+  def call
+    github_client = Octokit::Client.new(access_token: @scm_token,
+                                        api_endpoint: @event_subscription_payload[:api_endpoint])
+    # https://docs.github.com/en/rest/reference/repos#create-a-commit-status
+    github_client.create_status(@event_subscription_payload[:target_repository_full_name],
+                                @event_subscription_payload[:commit_sha],
+                                @state,
+                                status_options)
```

Comment:
Before refactoring and around this line we had:
`@workflow_run.save_scm_report_success(request_context) if @workflow_run.present?`

That line has gone, we should bring it back. But let's wait until PR #12799  is merged because it changes that code a bit.